sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/ObjectWithAlternativeContentCheck.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SonarHTML :: SonarQube Plugin
+ * Copyright (c) 2010-2019 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.sonar;
+
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.Node;
+import org.sonar.plugins.html.node.TagNode;
+import org.sonar.plugins.html.node.TextNode;
+
+@Rule(key = "S5264")
+public class ObjectWithAlternativeContentCheck extends AbstractPageCheck {
+
+  private static class ObjectEntry {
+
+    TagNode object;
+    boolean hasAlternativeContent;
+    
+    public ObjectEntry(TagNode object) {
+      this.object = object;
+      this.hasAlternativeContent = false;
+    }
+  }
+
+  private Deque<ObjectEntry> stack = new LinkedList<>();
```

Comment:
I am not sure you actually need a stack of `object` elements to be able to report issue. It seems that you only need stack which is 1 element deep, so you can just keep a field, where you set `null` in `endElement`. It will be bit simpler that way.

Reply:
Indeed. The stack was actually useful for an older version of the spec.